### ISSUE TITLE
Prevent UnusedVariable false positive when setting value on nested array

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#2336]
```php
function foo(array $arr): array {
    $c = "c";
    /** @psalm-suppress MixedArrayAssignment */
    $arr["a"]["b"][$c] = 1;
    return $arr;
}
```
Expected: No issue
Actual: UnusedVariable $c

seen in CakePHP cc @ADmad

**Comment by ADmad** [comment]
Thanks!
